File: ClientApp/src/app/model/Desk.ts
import { User } from '../user';

export interface Desk {
  id: Number;
  locationid: Number;
  bookedto: Date;
  bookedfrom: Date;
  isarchived: Boolean;
  bookby: User;
}


export interface Desklocation {
  id: Number;
  realDeskName: string;
  name: string;
  customerService: boolean;
  eccDesk: boolean;
  isbooked: boolean;
  desks: Desk[];
  floor: string;
}


export interface DeskDates {
  name: string;
  dates: Date;
}



export interface ResourceRoom {
  name: string;
  address: Date;
}



export interface Agreement {
  Id: Number;
  AcceptedDate: Date;
  UserName: string;

}

File: Modal/Desk.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace EusaHotDeskBooking.Modal
{
    public class Desk
    {

        public int Id { get; set; }


        [ForeignKey("location")]
        public int? locationId { get; set; }
        public virtual DeskLocation location { get; set; }

        public virtual User Bookby { get; set; }
        public DateTime? Bookedto { get; set; }
        public DateTime? Bookedfrom { get; set; }
        public bool Isarchived { get; set; }


    }



    public class DeskDates
    {
         public string Name { get; set; }

        public DateTime? Dates { get; set; }
    }
}